src/components/ButtonIcon/index.js

```diff
@@ -11,6 +14,14 @@ export default class ButtonIcon extends Component {
     constructor(props) {
         super(props);
         this.buttonRef = React.createRef();
+        this.showTooltip = this.showTooltip.bind(this);
+        this.hideTooltip = this.hideTooltip.bind(this);
+        this.handleMouseEnter = this.handleMouseEnter.bind(this);
+        this.handleMouseLeave = this.handleMouseLeave.bind(this);
+        this.windowScrolling = new WindowScrolling();
+        this.state = {
+            isTooltipVisible: false,
+        };
```

Comment:
here instead I think we should migrate this component to functional as it is simple, then you can use `src/components/InternalTooltip/hooks/useDefaultTooltipConnector.js` as we do in the InternalTooltip examples